File: PlatformaRecrutari.Dto/Sessions/FormDto.cs
﻿using PlatformaRecrutari.Dto.Sessions.FormQuesitons;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PlatformaRecrutari.Dto.Sessions
{
    public class FormDto
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public List<ShortQuestionDto> ShortQuestions { get; set; }
        public List<MultipleQuestionDto> MultipleQuestions { get; set; }
        public List<SelectBoxesQuestionDto> SelectBoxesQuestions { get; set; }
        public List<GridMultipleQuestionDto> GridMultipleQuestions { get; set; }
        public List<GridSelectBoxesQuestionDto> GridSelectBoxesQuestions { get; set; }
    }
}

File: PlatformaRecrutari.Web/ClientApp/src/shared/dto/form-dto.ts
import { GridMultipleQuestionDto } from "./questions/grid-multiple-question";
import { GridSelectBoxesQuestionDto } from "./questions/grid-select-boxes-question-dto";
import { MultipleQuestionDto } from "./questions/multiple-question-dto";
import { SelectBoxesQuestionDto } from "./questions/select-boxes-question-dto";
import { ShortQuestionDto } from "./questions/short-question-dto";

export interface FormDto {
  title: string;
  description: string;
  shortQuestions: ShortQuestionDto[];
  multipleQuestions: MultipleQuestionDto[];
  selectBoxesQuestions: SelectBoxesQuestionDto[];
  gridMultipleQuestions: GridMultipleQuestionDto[];
  gridSelectBoxesQuestions: GridSelectBoxesQuestionDto[];
}
